cloudmarker/alerts/emailalert.py

```diff
@@ -56,22 +62,58 @@ def done(self):
         this example implementation, we properly terminate the JSON
         array in the email body.
 
+        The connection is made based whether the SMTP or SMTP_SSL is
+        required which is determined by ``use_ssl`` param. The SMTP
+        login process is conducted if ``username`` param is provided.
+
         """
-        message = MIMEMultipart()
-        message['Date'] = email.utils.formatdate(localtime=True)
-        message['Subject'] = self.subject
-        message['From'] = self.sender
-        message['To'] = email.utils.COMMASPACE.join(self.to)
+        _message = MIMEMultipart()
+        _message['Date'] = email.utils.formatdate(localtime=True)
+        _message['Subject'] = self._subject
+        _message['From'] = self._sender
+        _message['To'] = email.utils.COMMASPACE.join(self._to)
```

Comment:
These local variables should not begin with an underscore. There is no notion of public vs. internal usage associated with local variables because, well, they are local. Therefore, the convention of beginning an identifier name with underscore does not apply here.

Did you read the style guide I mentioned earlier: [PEP-8](https://www.python.org/dev/peps/pep-0008/)?

It can be confusing to get this right without reading the style guide. I suggest reading the style guide once to understand the coding conventions used in the Python development community. Here are the relevant excerpts:

> ### [Function and Variable Names](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)
> Function names should be lowercase, with words separated by underscores as necessary to improve readability.
>
> Variable names follow the same convention as function names.

> ### [Method Names and Instance Variables](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)
> Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability.
>
> Use one leading underscore only for non-public methods and instance variables.
>
> To avoid name clashes with subclasses, use two leading underscores to invoke Python's name mangling rules.

Reply:
https://github.com/cloudmarker/cloudmarker/pull/75/commits/d092d76434f27c46450ee99c9fa9ed8ce75f0ba0#diff-28d4faa43f2d620f5a2c54123370ee47R72